### ISSUE TITLE
Fowarding headers from Supergraph to subgraphs

### ISSUE THREAD

**Issue by Shibbaz** [Gusto/apollo-federation-ruby#255]
Hi. 
That's my blocker while developing backend. Passing headers from supergraph to subgraphs.
Through logger I can observe that the node gateway is receiving headers but once I debug Rails graphql controller, It's not getting anything.
  graphql_controller.rb

       def execute
          variables = prepare_variables(params[:variables])
          query = params[:query]
          operation_name = params[:operationName]
          context = {
            current_user: current_user,
            session: session,
            tracing_enabled: ApolloFederation::Tracing.should_add_traces(request.headers),
      
          }
          result = UsersSchema.execute(query, variables: variables, context: context, operation_name: operation_name)
          render json: result
        rescue StandardError => e
          raise e unless Rails.env.development?
          handle_error_in_development(e)
        end

   gateway.ts
    
    const { ApolloServer, setContext } = require('apollo-server');
    const { ApolloGateway, RemoteGraphQLDataSource } = require('@apollo/gateway');
    const { readFileSync } = require('fs');
    const supergraphSdl = readFileSync('./supergraph.graphql').toString();
    import { ApolloLogPlugin } from 'apollo-log';
    
    const plugins = [ApolloLogPlugin];
    
    const gateway = new ApolloGateway({
        supergraphSdl,
    });
    
    
    const server = new ApolloServer({
        gateway,
        plugins: plugins,
        context: ({ req }) => {
            return {
              serverRequest: req,
            };
        }
    });
    
    server.listen().then(({ url }) => {
        console.log(`🚀 Gateway ready at ${url}`);
    }).catch(err => {console.error(err)});

**Comment by sethc2** [comment]
This is probably a better question for `https://github.com/apollographql/federation` as this is not related to the apollo-federation-ruby gem.

Apollo also has a support article detailing how it should work - https://support.apollographql.com/hc/en-us/articles/6135650062355-How-do-I-forward-headers-from-gateway-to-subgraph-.

You can try something like this perhaps.

```typescript
class AuthenticatedDataSource extends RemoteGraphQLDataSource {
  willSendRequest({ request, context }) { // this context is whatever you returned in constructing the ApolloServer
    request.http.headers.set("Authorization", context.authorizationHeader);
  }
}

const gateway = new ApolloGateway({
  supergraphSdl,
  buildService({ name, url }) {
    return new AuthenticatedDataSource({ url });
  },
});

const server = new ApolloServer({
    gateway,
    plugins,
    context: ({ req }) => {
        return {
          serverRequest: req,
          authorizationHeader: req.headers.authorization
        };
    }
});
```

**Comment by Shibbaz** [comment]
@sethc2 Yeah, I got it, that's line req.headers.authorization I needed, I came to this conclusion a moment before you posted, yet thanks :D
